x/ecocredit/marketplace/msg_ask_denom.go

```diff
@@ -0,0 +1,19 @@
+package marketplace
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+func (m AllowedDenom) Validate() error {
+	if err := sdk.ValidateDenom(m.BankDenom); err != nil {
+		return sdkerrors.ErrInvalidRequest.Wrapf("invalid denom: %s", err.Error())
+	}
+	if err := sdk.ValidateDenom(m.DisplayDenom); err != nil {
+		return sdkerrors.ErrInvalidRequest.Wrapf("invalid display_denom: %s", err.Error())
+	}
+	if m.Exponent > 18 {
```

Comment:
not sure if we need to guard this as its informational, im leaning towards removing

Reply:
Maybe we should validate for exponents which have an official SI unit.

Reply:
Maybe we should validate for exponents that have an official SI prefix.

Reply:
looks like we have a list in core/utils, using that for validation now 👍🏻